pkg/cephfs/controllerserver.go

```diff
@@ -285,3 +285,39 @@ func (cs *ControllerServer) ValidateVolumeCapabilities(
 		},
 	}, nil
 }
+
+// ExpandVolume expand CephFS Volumes on demand based on resizer request
+func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
+
+	if err := cs.validateExpandVolumeRequest(req); err != nil {
+		klog.Errorf("ControllerExpandVolumeRequest validation failed: %v", err)
+		return nil, err
+	}
+
+	volID := req.GetVolumeId()
+	secret := req.GetSecrets()
+
+	cr, err := util.NewAdminCredentials(secret)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	defer cr.DeleteCredentials()
+
+	volOptions, volIdentifier, err := newVolumeOptionsFromVolID(ctx, volID, nil, secret)
+
+	if err != nil {
+		klog.Errorf("validation and extraction of volume options failed: %v", err)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if err = createVolume(ctx, volOptions, cr, volumeID(volIdentifier.FsSubvolName), req.GetCapacityRange().GetRequiredBytes()); err != nil {
```

Comment:
Ideally, we don't want to use the `create subvolume` to resize. how urgent is this feature? We should use `fs subvolume resize` interface to resize CephFS subvolumes, https://github.com/ceph/ceph/pull/30054/

Reply:
why create volume is used here?

Reply:
@humblec We will have `fs subvolume resize` available in master by this weekend. But not sure about the next nautilus release date for you to use it. I'd suggest you to check with the Ceph Community Mailinglist about the date, so that you could take a decision.

[create_subvolume](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/volumes/fs/subvolume.py#L60) should allow you to resize the subvolume for now. Then it's better you have [this bugfix](https://tracker.ceph.com/issues/41884) in the version of nautilus you're using. This bug hits the exact code you're using now.

Reply:
> @ajarr this PR Is important, so I would rather stick to createVolume for now and we can accomodate fs subvolume resize whenever its available to consume.

Sounds OK

Reply:
:+1: